supervisor/api/backups.py

```diff
@@ -91,6 +92,14 @@ def _extract_slug(self, request):
             raise APIError("Backup does not exist")
         return backup
 
+    def _backup_location(self, backup: Backup) -> str | None:
```

Comment:
I think we should LRU this

Reply:
or even let backup make a property kinda `backup.location`

Reply:
Cached property is probably the best option, what do you think @mdegat01 ?

Reply:
Yeah good idea. Expensive to keep looking it up since we have to do path checking. And I don't really know of a better way to do it.

Small note though, we can use [`is_relative_to`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_relative_to) instead of string checking